packages/admin/admin/src/dataGrid/DataGridColumnsManagement.tsx

```diff
@@ -0,0 +1,92 @@
+// MUI's implementation: https://github.com/mui/mui-x/blob/v7.x/packages/x-data-grid/src/components/columnsManagement/GridColumnsManagement.tsx
+import {
+    type ComponentsOverrides,
+    type List as MuiList,
+    type ListItem as MuiListItem,
+    type Switch as MuiSwitch,
+    type Theme,
+    type Typography,
+    useThemeProps,
+} from "@mui/material";
+import { gridColumnVisibilityModelSelector, useGridApiContext, useGridSelector } from "@mui/x-data-grid-pro";
+import { type ChangeEvent } from "react";
+
+import { type ThemedComponentBaseProps } from "../helpers/ThemedComponentBaseProps";
+import { Body, List, ListItem, ListItemTitle, Root, Switch } from "./DataGridColumnsManagement.sc";
+
+export type DataGridColumnsManagementProps = ThemedComponentBaseProps<{
+    root: "div";
+    body: "div";
+    switch: typeof MuiSwitch;
+    list: typeof MuiList;
+    listItem: typeof MuiListItem;
+    listItemTitle: typeof Typography;
+}>;
+
+export type DataGridColumnsManagementClassKey = "root" | "body" | "list" | "listItem" | "switch" | "listItemTitle";
+
+export const DataGridColumnsManagement = (inProps: DataGridColumnsManagementProps) => {
+    const { sx, className, slotProps = {} } = useThemeProps({ props: inProps, name: "CometAdminDataGridColumnsManagement" });
+
+    const apiRef = useGridApiContext();
+    const columnVisibilityModel = useGridSelector(apiRef, gridColumnVisibilityModelSelector);
+    const columns = apiRef.current.getAllColumns();
+
+    const toggleColumn = (event: ChangeEvent<HTMLInputElement>) => {
+        const { name: field } = event.target;
+        const column = columns.find((value) => value.field === field);
+        if (column?.hideable) {
+            const isCurrentlyVisible = columnVisibilityModel[field] !== false;
+
+            const newVisibilityModel = {
+                ...columnVisibilityModel,
+                [field]: !isCurrentlyVisible,
+            };
+
+            apiRef.current.setColumnVisibilityModel(newVisibilityModel);
+        }
+    };
+
+    return (
+        <Root sx={sx} className={className} {...slotProps.root}>
+            <Body {...slotProps.body}>
+                <List {...slotProps.list}>
+                    {columns.map((column, index) => {
+                        const checked = columnVisibilityModel?.[column.field] ?? true;
+                        return (
+                            <ListItem key={`columns-management-row${index}`} {...slotProps.listItem}>
```

Comment:
I don't think the prefix is necessary 🤔 

```suggestion
                            <ListItem key={index} {...slotProps.listItem}>
```

Reply:
I'd prefer using `column.field` instead of `index`, which needs to be unique.

Reply:
probably a bit edgy, but a problem with `column.field` as key could be that, `column.field` must not be unique.